enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,416 @@
+---
+title: Supporting out-of-tree drivers on BareMetal & OpenShift
+  - "@zvonkok"
+reviewers:
+  - TBD
+  - "@mrunalp"
+approvers:
+  - TBD
+  - "@mrunalp"
+creation-date: 2020-04-03
+last-updated: 2020-06-02
+status: provisional
+see-also:
+  - "/enhancements/TODO.md"  
+replaces:
+  - "/enhancements/TODO.md"
+superseded-by:
+  - "/enhancements/TODO.md"
+---
+
+# Supporting out-of-tree drivers on BareMetal & OpenShift
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. Are users willing to provide precompiled DriverContainers (We have already some commitments)?  This helps immensely in disconnected environments and only validated combinations of driver + kernel would be e.g. provided. Source instal should be the fallback solution
+
+## Summary
+
+Support out-of-tree drivers on OpenShift for enabling custom hardware and third-party kernel modules on container and non-container based operating systems. 
+
+## Motivation
+
+
+In OpenShift v3.x out-of-tree drivers could be easily installed on the nodes, since the node was a full RHEL node with a subscription and needed tools were installed easily with yum. 
+In OpenShift v4.x this changed with the introduction of RHCOS. There are currently two different documented ways to enable out-of-tree drivers. One way is using a SRO based operator such as the nvidia GPU operator and the other is using kmods-via-containers. 
+We want to come up with a unified solution that works for our customers across RHCOS and RHEL. The solution should also help customers that are currently on RHEL7 to consider moving to RHCOS which is fully managed and easier to support in OpenShift.
+ 
+#### Day-1 DriverContainer Management  [OpenShift | BareMetal]
+ 
+TODO: LONGER TERM
+ 
+#### Day-2 DriverContainer Management OpenShift
+ 
+More and more customers/partners want to enable hardware and or software on OpenShift that need kernel drivers, which are currently not (and may never be) upstream   and out-of-tree. 
+We want to provide a unified way to support out-of-tree kernel drivers (multiple) on OpenShift for day-1 (long term) and day-2 (short term). It has to work for classical (RHEL7, 8, Fedora) and container based operating systems (RHCOS, FCOS) in the same way. 
+Many customers/partners are using dkms/akmods as the solution to build, rebuild modules on kernel changes. Adopting dkms is not a solution; we need to  create an own acceptable solution for RHEL and Fedora. 
+ 
+#### Fill the gap of providing drivers that are not yet upstream or never be
+
+For the DriverContainer we need to cover several stages of driver packaging. 
```

Comment:
i know DriverContainer is a concept from SRO, it helps to provide some definitions i think or links for others.

Reply:
i see terminology is defined at the bottom.  this is easier to follow by moving some of that content up to the top.

Reply:
There is a terminology section, maybe I move it up, before the proposal section.

Reply:
I moved it up for clarification before going too much into detail.